### ISSUE TITLE
Request: Smart Firing Platforms as drones

### ISSUE THREAD

**Issue by aga9** [chummer5a/chummer5a#2860]
### Environment
Chummer Version: 5.201.0
Environment: Windows 10
Runtime: .NET 4.5.1

### Expected behaviour
Smart Firing Platforms have their own initiative, stats, and can run program. They are more drones than weapon accessories for all intents.

They should be listed as a drone, maybe as an optional rule. As drones, they will be able to swap weapon, load programs, and have stats that weapon mods cannot.

### Actual behaviour
Smart Firing Platforms are weapon mods that cannot load programs.

**Comment by Haplo164** [comment]
Should be able to treat it like cyberhand/eye drones and add a smartfiring platform to the vehicle list. Getting the weapon to appear may be tricky though.

**Comment by aga9** [comment]
What do you mean by "getting the weapon to appear"?

If you mean to get a weapon in gear section to also appear on the vehicle's weapon mount:

I don't think that would be necessary. A simple drone setup should suffice.

**Comment by chummer5a** [comment]
Haplo's idea is that when you'd add the smart firing platform to a weapon, it would also add it to the vehicles panel with the weapon mounted. Saves having to duplicate the weapon. Not exceptionally difficult to implement, but awkward. 

Easier to just make a drone vehicle, add a custom weapon mount type that requires and consumes an Under mount and have done with it.

**Comment by Haplo164** [comment]
I guess the real issue here is that the smart firing platform is basically a drone, but the weapon can also be used by metahumans with the platform attached but not deployed.

**Comment by aga9** [comment]
Took a crack at this:
[amend_vehicles.txt](https://github.com/chummer5a/chummer5a/files/3125825/amend_vehicles.txt)

**Comment by chummer5a** [comment]
Needs a weapon mount entry, category would probably be small?

**Comment by aga9** [comment]
Added weapon mounts. I used the "Standard [SR5]" mount since SMP is also from CRB.

There's a bit of a rules problem there, since RAW SMP is an underbarrel mod, thus nothing smaller than a rifle can be mounted on an SMP. As such, the SMP mount _cannot_ be _small_. But I think most people would agree that's a bit silly.

[amend_vehicles.txt](https://github.com/chummer5a/chummer5a/files/3138508/amend_vehicles.txt)

One question: I've used the same <id> as existing Smart Firing Platform. Is that okay?
